src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -62,7 +62,7 @@ class PythonToolRequirementsBase(Subsystem, ExportableTool):
     #  requirements that reflect any minimum capabilities Pants assumes about the tool.
     default_requirements: Sequence[str] = []
 
-    default_interpreter_constraints: ClassVar[Sequence[str]] = ["CPython>=3.7,<4"]
+    default_interpreter_constraints: ClassVar[Sequence[str]] = ["CPython>=3.8,<4"]
```

Comment:
This feels really heavy-handed, but I couldn't figure out how to get the remaining tests passing without enforcing this across the board for all Python tools.